container_workflow_tool/distgit.py

```diff
@@ -538,16 +553,14 @@ def is_git(x): return os.path.isdir(os.path.join(x, '.git'))
         # Walk through the repositories and show changes made in the last commit
         for path in files:
             repo = Repo(path)
-            branch = repo.active_branch.name
-            # Get a list of commits that have not been pushed to remote
-            select = "origin/" + branch + ".." + branch
-            commits = [i for i in repo.iter_commits(select)]
             # Only show changes if there are unpushed commits to show
             # or we only want the diff of unstaged changes
-            if commits or diff:
+            if self._get_unpushed_commits(repo) or diff:
                 # Clears the screen
                 print(chr(27) + "[2J")
                 # Force pager for short git diffs
                 subprocess.run("git config core.pager 'less -+F' --replace-all", cwd=path, shell=True)
                 # Not using GitPython as its git.show seems to have some problems with encoding
                 subprocess.run(['git', command], cwd=path)
+            else:
+                self.logger.info(f"There are no unpushed commits for `git {command}`.")
```

Comment:
Ah sorry, I forgot the clause could be added here as well, but it does not make much sense for this part as the only time this will actually be readable (without scrolling in the terminal) is when there are no changes for any repository.

Reply:
The question is if we want to inform the user, that there were no unpushed changes or just or just silently skip it.

Reply:
I myself would skip it, as it will be missed by the user anyway, unless we force it into some pager similarly to the show/diff.

Reply:
Or if the logger is redirected to something other than stdin I guess